lib/Expr/BitfieldSimplifier.cpp

```diff
@@ -99,21 +97,21 @@ BitfieldSimplifier::ExprBitsInfo BitfieldSimplifier::doSimplifyBits(ref<Expr> e,
     /* Apply kind-specific knowledge to obtain knownBits for e and
        ignoredBits for kids of e, then to optimize e */
     switch (e->getKind()) {
-        // TODO: Concat, Read, AShr
+            // TODO: Concat, Read, AShr
 
         case Expr::And:
             rbits.knownOneBits = bits[0].knownOneBits & bits[1].knownOneBits;
             rbits.knownZeroBits = bits[0].knownZeroBits | bits[1].knownZeroBits;
 
             bits[0].ignoredBits = ignoredBits | bits[1].knownZeroBits;
-            bits[1].ignoredBits = ignoredBits | (bits[0].knownZeroBits & ~bits[1].knownZeroBits);
+            bits[1].ignoredBits = ignoredBits | bits[0].knownZeroBits;
```

Comment:
Why is the remaining part of the expression missing?

Reply:
If a bit on the left side is 0, doesn't it mean that we can ignore the corresponding bit on the right side? And is also commutative so wouldn't it make sense to have the same expression for ignored bits on the left and on the right?

Reply:
Could be, I'll have a better look later. In the meantime, could you move optimizations/bugfixes in a separate commit? The first commit would just have the mechanical conversion to APInt. This would make it much simpler to review.

Reply:
In the original zeroMask is only needed because the masks are 64 bits wide even when the expressions are narrower than 64 bits. APInts can be any number of bits wide so that would not be needed anymore. Should I keep 64-bit masks and zeroMask in the mechanical conversion or not?

Reply:
Yes, please keep 64-bits.